packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,367 @@
+import 'package:vector_graphics_compiler/src/svg/node.dart';
+import 'package:vector_graphics_compiler/src/svg/parser.dart';
+import 'package:vector_graphics_compiler/src/svg/resolver.dart';
+import 'package:vector_graphics_compiler/src/svg/visitor.dart';
+import 'dart:core';
+import 'dart:typed_data';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart';
+import 'package:path_ops/path_ops.dart' as path_ops;
+
+class _EmptyNode extends Node {
+  const _EmptyNode();
+
+  @override
+  S accept<S, V>(Visitor<S, V> visitor, V data) {
+    return visitor.visitEmptyNode(this, data);
+  }
+
+  @override
+  void visitChildren(NodeCallback visitor) {}
+}
+
+// TODO: Edit result class to pass children information to parent
+class _Result {
+  _Result(this.node);
+
+  final Node node;
+  int childCount = 0;
```

Comment:
I think this should get deleted and just use `children.length`.

Also, children should be final.

Reply:
fixed.